### PR TITLE
Exists query

### PR DESCRIPTION
**Pull request by limick**
This addresses #1069 by adding an `exists` function to the `PersistQueryRead` type class, which already provides the `count` function. The `count` function will be used for a fallback implementation for backends that don't provide a separate existence check.

Before submitting your PR, check that you've:

- [X] Bumped the version number (can we get this into 2.11?)
- [X] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html)
- [X] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddock

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [x] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

File: persistent-mongoDB/ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog for persistent-mongoDB
 
+## 2.11.0
+
+* Naive implementation of `exists`function from `PersistQueryRead` type class using `count`.
```
Comment (MaxGabriel) [inline persistent-mongoDB/ChangeLog.md:5]:
```suggestion
* Naive implementation of `exists` function from `PersistQueryRead` type class using `count`.
```

File: persistent-sqlite/ChangeLog.md [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 * [#1060](https://github.com/yesodweb/persistent/pull/1060)
   * The QuasiQuoter now supports `OnDelete` and `OnUpdate` cascade options.
+* [#1069](https://github.com/yesodweb/persistent/pull/1070)
```
Comment (MaxGabriel) [inline persistent-sqlite/ChangeLog.md:7]: Is this 1069/1070 mismatch intentional?

Comment (MaxGabriel) [reply persistent-sqlite/ChangeLog.md:7]: (This also applies to the other changelog files)

File: persistent/Database/Persist/Sql/Orphan/PersistQuery.hs [hunk 3]
```diff
@@ -55,6 +55,31 @@ instance PersistQueryRead SqlBackend where
...
+        let sql = mconcat
+                [ "SELECT EXISTS(SELECT 1 FROM "
+                , connEscapeName conn $ entityDB t
+                , wher
+                , ")"
+                ]
+        withRawQuery sql (getFiltsValues conn filts) $ do
+            mm <- CL.head
+            case mm of
+              Just [PersistBool b]  -> return b
```
Comment (MaxGabriel) [inline persistent/Database/Persist/Sql/Orphan/PersistQuery.hs:72]:
```suggestion
              Just [PersistBool b]  -> return b -- Postgres
```

Comment (MaxGabriel) [inline persistent/Database/Persist/Sql/Orphan/PersistQuery.hs:72]:
```suggestion
              Just [PersistBool b]  -> return b -- Postgres
```

File: persistent/Database/Persist/Sql/Orphan/PersistQuery.hs [hunk 4]
```diff
@@ -55,6 +55,31 @@ instance PersistQueryRead SqlBackend where
...
+              Just [PersistInt64 i] -> return $ i > 0
```
Comment (MaxGabriel) [inline persistent/Database/Persist/Sql/Orphan/PersistQuery.hs:73]:
```suggestion
              Just [PersistInt64 i] -> return $ i > 0 -- MySQL
              -- The following haven't been tested, but this is based off the code used in count
```

File: persistent/Database/Persist/Sql/Orphan/PersistQuery.hs [hunk 5]
```diff
@@ -55,6 +55,31 @@ instance PersistQueryRead SqlBackend where
...
+              Just [PersistDouble i] -> return $ (truncate i :: Int64) > 0 -- gb oracle
+              Just [PersistByteString i] -> case readInteger i of -- gb mssql
+                                              Just (ret,"") -> return $ ret > 0
+                                              xs -> error $ "invalid number i["++show i++"] xs[" ++ show xs ++ "]"
+              Just xs -> error $ "count:invalid sql  return xs["++show xs++"] sql["++show sql++"]"
+              Nothing -> error $ "count:invalid sql returned nothing sql["++show sql++"]"
```
Comment (MaxGabriel) [inline persistent/Database/Persist/Sql/Orphan/PersistQuery.hs:79]:
```suggestion
              Just xs -> error $ "exists:invalid sql  return xs["++show xs++"] sql["++show sql++"]"
              Nothing -> error $ "exists:invalid sql returned nothing sql["++show sql++"]"
```

**Comment by MaxGabriel** [comment]
This looks good to me, made a few suggested changes

**Comment by parsonsmatt** [comment]
@limick Would you mind applying the relevant suggestions? Should be doable from the web editor. THen we can get this merged in and released :smile:

File: persistent-sqlite/ChangeLog.md (hunk 2 above)
Comment (limick) [reply persistent-sqlite/ChangeLog.md:7]: I think I somehow came to the conclusion that the first is the issue number while the second is the PR, but I'll change it to the PR everywhere.

**Comment by limick** [comment]
Sure, thanks for your patience.
I no longer have access to the original repository, so I created a new PR #1131 to replace this one. I've applied the suggestions there (thanks!) and rebased onto the destination branch.

**Pull request closed without merging**